lib/datadog/grok/src/grok_filter.rs

```diff
@@ -153,5 +158,123 @@ pub fn apply_filter(value: &Value, filter: &GrokFilter) -> Result<Value, GrokRun
                 value.to_string(),
             )),
         },
+        GrokFilter::Date(format, tz_regex_opt, target_tz, with_tz) => match value {
+            Value::Bytes(bytes) => {
+                let value = String::from_utf8_lossy(bytes);
+                match tz_regex_opt {
+                    Some(re) => {
+                        let tz = re
+                            .captures(&value)
+                            .ok_or_else(|| {
+                                GrokRuntimeError::FailedToApplyFilter(
+                                    filter.to_string(),
+                                    value.to_string(),
+                                )
+                            })?
+                            .name("tz")
+                            .unwrap()
```

Comment:
So, I'm wondering if this `unwrap` is safe.  I think this will panic if there is no `tz` group in the regex, which will only be added if `capture_tz` is true in `time_format_to_regex`. It's not completely clear if this will always be the case here.

If it is totally safe I think it would be worth changing it to `expect` and adding a comment to explain why it is safe.

Reply:
It's safe, because this regex will always contain a tz capture(that is its only purpose). I've tried to make it more clear with little refactoring and more comments.